### PR TITLE
Update coverage invocation

### PR DESCRIPTION
**Pull request by khk-globus**

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [tox]
-envlist = cov-clean,py{310,37}-nodeps,py{310,39,38,37},cov-report
+envlist = py{37,38,39,310}
```
Comment (sirosen) [inline tox.ini:2]: The `-nodeps` builds guard against introduction of hard dependencies on third party libraries.

That makes it safer and easier to maintain a library which _allows_ integration with `redis`, `boto3`, and potentially other libraries, but doesn't require them to be present in order to be importable.

Removing this testing may impact the safety of relying on `funcx-common` in client-side software like `funcx` and `funcx-endpoint`, so I would recommend very strongly against doing this.

File: tox.ini [hunk 2]
```diff
@@ -11,25 +11,15 @@ passenv =
     AWS_SECRET_ACCESS_KEY
     AWS_SESSION_TOKEN
 usedevelop = true
+deps =
+    redis
+    coverage
```
Comment (sirosen) [inline tox.ini:16]: Removing the `!nodeps` factorized installation of `boto3` and `moto` means that tests requiring boto3+moto will never run.

**Review by sirosen — CHANGES_REQUESTED** [review]
This PR does not describe itself accurately. It has numerous effects other than a simple switch from `pytest-cov` to `coverage`.

If `pytest-cov` is being removed -- which would be fine -- then the installable extras need to be updated in `setup.cfg`:
https://github.com/funcx-faas/funcx-common/blob/d3841ac364e6774cc5fc5d413a0337c171e6588b/setup.cfg#L35

**Comment by khk-globus** [comment]
Temporarily closing until I get back to this.

**Pull request closed without merging**